tic_tac_toe/game.py

```diff
@@ -63,42 +108,137 @@ def _check_winning_combinations(board, player):
+        
+    return combination
+    """
+
+   
 
 # public interface
 def start_new_game(player1, player2):
     """
     Creates and returns a new game configuration.
     """
-    pass
-
+    game = {
+            'player1': player1,
+            'player2': player2,
+            'board': [
+                ["-", "-", "-"],
+                ["-", "-", "-"],
+                ["-", "-", "-"],
+            ],
+            'next_turn': player1,
+            'winner': None
+        }
+    return game
 
 def get_winner(game):
     """
     Returns the winner player if any, or None otherwise.
     """
-    pass
-
+    return game['winner']
+    
 
 def move(game, player, position):
     """
     Performs a player movement in the game. Must ensure all the pre requisites
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    next_player = get_next_turn(game)
+    if _position_is_valid(position) != True:
+        raise InvalidMovement("Position out of range.")
+    elif _position_is_empty_in_board(position, board) != True:
+        raise InvalidMovement("Position already taken.")
+    elif player != next_player:
+        raise InvalidMovement((next_player, " moves next"))
+    else:
+        x = position[0]
+        y = position[1]
+        game['board'][x][y] = player
+    if player == "X":
+        game['next_turn'] = "O"
+    else:
+        game['next_turn'] = "X"
+    game['winner'] = _check_winning_combinations(board, player)
+    winner = get_winner(game)
+    if winner != None:
+        raise GameOver('"{}" wins!'.format(winner))
+    if _board_is_full(board) == True:
+        raise GameOver("Game is tied!")
+    
+
+   # raise  GameOver((player, " wins!"))     
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
-
+    board = game['board']
+    image = '\n'
+    image += "{}  |  {}  |  {}\n".format(board[0][0],board[0][1],board[0][2])
+    image += "--------------\n"
+    image += "{}  |  {}  |  {}\n".format(board[1][0],board[1][1],board[1][2])
+    image += "--------------\n"
+    image += "{}  |  {}  |  {}\n".format(board[2][0],board[2][1],board[2][2])
+ 
+    return image
 
 def get_next_turn(game):
     """
     Returns the player who plays next, or None if the game is already over.
     """
-    pass
+    return game['next_turn']
+
+
+
+"""
+game = {
+    'player1': "X",
+    'player2': "O",
+    'board': [
+        ["-", "-", "X"],
+        ["-", "-", "X"],
+        ["-", "-", "-"],
+    ],
+    'next_turn': "X",
+    'winner': None
+}
+"""
+#print(_position_is_empty_in_board((0,0), game['board']))
+
+#print(_board_is_full(game['board']))
+#combination = _check_winning_combinations(game['board'], 'X')
+#print(_is_winning_combination(game['board'], combination, 'X'))
+#player1="X"
+##player2="O"
+#move(game,player1,(2,2))
+
```

Comment:
Great job all around! I really appreciate that you included some of your old approaches in multi-line comments. Since we're generally not making tons of commits it's a good way to see and show process. Our group definitely took a few approaches that didn't work out in the end.